situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java

```diff
@@ -152,7 +156,11 @@ public void stop() {
   private void createIndexTemplate() throws IOException {
     final String indexAlias = esSinkConfig.getIndexConfiguration().getIndexAlias();
     final PutIndexTemplateRequest putIndexTemplateRequest = new PutIndexTemplateRequest(indexAlias + "-index-template");
-    putIndexTemplateRequest.patterns(Collections.singletonList(indexAlias + "-*"));
+    if (indexType.equals(IndexConstants.RAW)) {
+      putIndexTemplateRequest.patterns(Collections.singletonList(indexAlias + "-*"));
+    } else {
+      putIndexTemplateRequest.patterns(Collections.singletonList(indexAlias));
+    }
```

Comment:
nit: This code block and the if in Line#179 makes the sink code imperfect. I think we can do all the if else check on ```IndexConstants.Raw``` inside the indexConfiguration. You should consider this later.

Reply:
So if I understand correctly, we can refactor index pattern and name into IndexConfig, right? Makes sense.